force-app/main/default/classes/DMLOperations.cls

```diff
@@ -213,8 +305,22 @@ public with sharing class DMLOperations {
      * @param numOfCases The number of cases to create.
      */
     public static void createAndDeleteCases(String accountId, Integer numOfCases) {
+        //generate list to store cases
+        list<Case> storeCase = new List<Case>();
+
+        for (Integer i = 0; i < numOfCases; i++){
+            Case caseToAdd = new Case();
+            storeCase.add(caseToAdd);
+            caseToAdd.AccountId = accountId;
```

Comment:
Try to reverse these two lines. Assign the AccountId before you add it to the list. Your way probably works too, just not accustomed to seeing it reversed.